logagg/forwarder.py

```diff
@@ -86,12 +88,29 @@ def _ack_messages(self, msgs):
                 msg.fin()
             except (SystemExit, KeyboardInterrupt): raise
             except:
-                self.log.exception('msg ack failed')
+                self.log.exception('msg_ack_failed')
 
-    def _write_messages(self, msgs):
-        # FIXME: what if a target fails to handle logs properly and
-        # throws an exception? we need to protect against that case.
-        for one_target in self.targets:
-            util.start_daemon_thread(one_target.handle_logs(msgs)).join
-            self.log.debug('Starting to write messeges ', target=one_target)
+    def _send_msgs_to_target(self, target, msgs):
+        while 1:
+            try:
+                target.handle_logs(msgs)
+                break
+            except (SystemExit, KeyboardInterrupt): raise
+            except:
+                # FIXME: need to check whether this works at all
+                # FIXME: do we log the failed messages themselves somewhere?
+                self.log.exception('_send_msgs_to_target_failed',
+                        target=target.__name__, num_msgs=len(msgs))
+                time.sleep(5) # FIXME: don't hard code this number
```

Comment:
need to remove hardcoded number and parametrize at some level (command-line arg or class level const etc)